### PR TITLE
IGAPP-498: Show snackbar if language change fails

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Review by Taggotty — APPROVED** [review]

**Review by SteffiStuffel — APPROVED** [review]
Tested and it works, if i add a 'throw new Error'. 

Should the console.warns('Route was not ready when translating..') in morphContantantLanguage maybe also be displayed in the Snackbar?

**Comment by klinzo** [comment]
> Tested and it works, if i add a 'throw new Error'.
> 
> Should the console.warns('Route was not ready when translating..') in morphContantantLanguage maybe also be displayed in the Snackbar?

@SteffiStuffel not sure about that, never had that warning yet and it would be a separate issue. Is it possible that this warning does only occur because of the throw new error you added?

**Comment by SteffiStuffel** [comment]
> > Tested and it works, if i add a 'throw new Error'.
> > Should the console.warns('Route was not ready when translating..') in morphContantantLanguage maybe also be displayed in the Snackbar?
> 
> @SteffiStuffel not sure about that, never had that warning yet and it would be a separate issue. Is it possible that this warning does only occur because of the throw new error you added?

No this warning is never showed in the Snackbar, my suggestion was that it maybe would make sense to show it in a snackbar if it occurs?

**Comment by klinzo** [comment]
As discussed: As long as nobody every experiences this it is not relevant.

File: native/src/modules/endpoint/sagas/watchContentLanguageSwitch.js [hunk 1]
```diff
@@ -75,10 +74,8 @@ export function* switchContentLanguage(
     }
     yield put(insert)
   } catch (e) {
-    if (fromError(e) === ErrorCodes.NetworkConnectionFailed) {
-      // TODO IGAPP-498 The alert should be replaced with a snackbar, hence the TFunction should also be removed.
-      Alert.alert(t('languageSwitchFailedTitle'), t('languageSwitchFailedMessage'))
```
Comment (maxammann) [inline native/src/modules/endpoint/sagas/watchContentLanguageSwitch.js:80]: Should we remove this translation?

**Review by maxammann — APPROVED** [review]

File: native/src/modules/endpoint/sagas/watchContentLanguageSwitch.js (hunk 1 above)
Comment (klinzo) [reply native/src/modules/endpoint/sagas/watchContentLanguageSwitch.js:80]: In general yes, but does this even work if the translations for malte are reimported afterwards?

**Pull request merged**